### ISSUE TITLE
Support Pickle protocol

### ISSUE THREAD

**Issue by SuperQ** [prometheus/graphite_exporter#76]
It would be useful to support the Graphite [Pickle protocol](https://graphite.readthedocs.io/en/latest/feeding-carbon.html#the-pickle-protocol).

Looks like there is a [Go library](https://github.com/lomik/graphite-pickle) that does this already. Shouldn't be to hard to implement.

**Comment by matthiasr** [comment]
yes, that seems like a really useful. This would be a separate port, right?

**Comment by brian-brazil** [comment]
Does anyone actually use this? I vaguely remember looking into it, and basically noone did so I never implemented it.

**Comment by SuperQ** [comment]
Yes, it would live on a different port.

Someone posted on IRC that they needed it, said they would make an attempt at adding it. I filed the issue for tracking.

**Comment by hhsnow** [comment]
One example user is Druid, which defaults to the pickle format. https://druid.apache.org/docs/latest/development/extensions-contrib/graphite.html. Producing/ingesting metrics in pickled batches seems to be more efficient, but the plaintext protocol option is available.

**Comment by rjshrjndrn** [comment]
I roamed a lot with the parsing error and came across this thread. Yeah adding picke support will be huge help @brian-brazil
